Sources/Apodini/Request/DynamicProperty.swift

```diff
@@ -0,0 +1,214 @@
+                case let target as Target:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "\(Target.self) \(property.name) on element \(info.name) must be a struct")
+                    
+                    operation(target, property.name)
+                case let dynamicProperty as DynamicProperty:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "DynamicProperty \(property.name) on element \(info.name) must be a struct")
+
+                    dynamicProperty.execute(operation)
+                case let dynamics as Traversible:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "Dynamics \(property.name) on element \(info.name) must be a struct")
+                
+                    dynamics.execute(operation)
+                default:
+                    break
+                }
+            }
+        } catch {
+            fatalError("Executing operation on all properties of \((try? typeInfo(of: Target.self))?.name ?? "Unknown Type") on DynamicProperty \((try? typeInfo(of: Self.self))?.name ?? "Unknown Type") failed.")
+        }
+    }
+    
+    mutating func apply<Target>(_ mutation: (inout Target, _ name: String) -> Void) {
+        do {
+            let info = try typeInfo(of: Self.self)
+
+            for property in info.properties {
+                let child = try property.get(from: self)
+
+                switch child {
+                case var target as Target:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "\(Target.self) \(property.name) on element \(info.name) must be a struct")
+                    
+                    mutation(&target, property.name)
+                    try property.set(value: target, on: &self)
+                case var dynamicProperty as DynamicProperty:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "DynamicProperty \(property.name) on element \(info.name) must be a struct")
+                    
+                    dynamicProperty.apply(mutation)
+                    try property.set(value: dynamicProperty, on: &self)
+                case var dynamics as Traversible:
+                    assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "Dynamics \(property.name) on element \(info.name) must be a struct")
+                
+                    dynamics.apply(mutation)
+                    try property.set(value: dynamics, on: &self)
+                default:
+                    break
+                }
+            }
+        } catch {
+            fatalError("Applying operation on all properties of \((try? typeInfo(of: Target.self))?.name ?? "Unknown Type") on DynamicProperty \((try? typeInfo(of: Self.self))?.name ?? "Unknown Type") failed.")
+        }
+    }
+}
+
+
+// MARK: Dynamics Implemenation
+
+private protocol Traversible {
```

Comment:
I would suggest to move these requirements to `Injectable` or `RequestInjectable` as one constraint that must be implemented by a type that can be `Injectable` or hide the functionality in a single `inject` method that does all the work for all types conforming to `Injectable` or `RequestInjectable`.  `DynamicProperty` or `DynamicParameter` would be an extension of that functionality.

Reply:
I would also like to have a bit more abstraction inside `DynamicsProperty.swift`, but I would not combine this with `RequestInjectable `. `RequestInjectable ` is a `Target`, not the infrastructure, not `Traversible`. Where would e.g. `Parameter` (which is the primary `RequestInjectable`) find the `Target`?